packages/dd-trace/src/dogstatsd.js

```diff
@@ -31,14 +41,31 @@ class Client {
     this._add(stat, value, 'c', tags)
   }
 
-  flush () {
+  flush (cb) {
     const queue = this._enqueue()
 
     if (this._queue.length === 0) return
 
     this._queue = []
 
-    if (this._family !== 0) {
+    if (this._httpOptions) {
+      requestQueue(queue, this._httpOptions, err => {
+        if (err) {
+          log.debug('HTTP error from agent: ' + err.stack)
+          if (err.status) {
+            // Inside this if-block, we have connectivity to the agent, but
+            // we're not getting a 200 from the proxy endpoint. Fall back to
+            // UDP and try again.
+            this._httpOptions = null
+            this._queue = queue
+            this.flush()
+          }
+        }
+        if (cb) {
+          cb(err)
```

Comment:
Should this also be logged?

Reply:
Yeah, and we're doing that a few lines up. I'll change the log level there to error.